src/Entity/InformationProduct.php

```diff
@@ -0,0 +1,286 @@
+     */
+    private $title;
+
+    /**
+     * The creators of the Information Product.
+     *
+     * @var string
+     *
+     * @ORM\Column(type="text")
+     *
+     * @Assert\NotBlank(
+     *     message="A creator is required."
+     * )
+     */
+    private $creators;
+
+    /**
+     * The publisher of the Information Product.
+     *
+     * @var string
+     *
+     * @ORM\Column(type="text")
+     *
+     * @Assert\NotBlank(
+     *     message="A publisher is required."
+     *
+     */
+    private $publisher;
+
+    /**
+     * An external DOI for the Information Product.
+     *
+     * @var string
+     *
+     * @ORM\Column(type="text", nullable=true)
+     */
+    private $externalDoi;
+
+    /**
+     * Is the Information Product published.
+     *
+     * @var boolean
+     *
+     * @ORM\Column(type="boolean")
+     */
+    private $published = false;
+
+    /**
+     * Is the Information Product remotely hosted.
+     *
+     * @var boolean
+     *
+     * @ORM\Column(type="boolean")
+     */
+    private $remoteResource = false;
+
+    /**
+     * The research groups this Information Product is associated with.
+     *
+     * @var Collection
+     *
+     * @ORM\ManyToMany(targetEntity=ResearchGroup::class)
+     */
+    private $researchGroups;
+
+    /**
+     * Constructor.
+     *
+     * Initializes collections to empty collections.
+     */
+    public function __construct()
+    {
+        $this->researchGroups = new ArrayCollection();
+    }
+
+    /**
+     * Get the title for this Information Product.
+     *
+     * @return string|null
+     */
+    public function getTitle(): ?string
+    {
+        return $this->title;
+    }
+
+    /**
+     * Set the title for this Information Product.
+     *
+     * @param string|null $shortTitle The title for this Information Product.
+     *
+     * @return self
+     */
+    public function setTitle(string $title): self
+    {
+        $this->title = $title;
+
+        return $this;
+    }
+
+    /**
+     * Get the creators for this Information Product.
+     *
+     * @return string|null
+     */
+    public function getCreators(): ?string
+    {
+        return $this->creators;
+    }
+
+    /**
+     * Set the creators of the Information Product.
+     *
+     * @param string $creators The creators of the Information Product.
+     *
+     * @return self
+     */
+    public function setCreators(string $creators): self
+    {
+        $this->creators = $creators;
+
+        return $this;
+    }
+
+    /**
+     * Get the publisher of the Information Product.
+     *
+     * @return string|null
+     */
+    public function getPublisher(): ?string
+    {
+        return $this->publisher;
+    }
+
+    /**
+     * Set the publisher of the Information Product.
+     *
+     * @param string $publisher
+     * @return self
+     */
+    public function setPublisher(string $publisher): self
+    {
+        $this->publisher = $publisher;
+
+        return $this;
```

Comment:
Do we need setters to return ?

Reply:
The make command does it, we are not consistent in other classes. I kinda like it, but we can remote it?

Reply:
I think from now on, we can add returns to setters